### PR TITLE
Jim/76903/money-component-ts-migration

### PR DESCRIPTION
**Pull request by jim-deriv**
## Changes:

Please include a summary of the change and which issue is fixed below:
-   ...

## When you need to add unit test

-   [ ] If this change disrupt current flow
-   [ ] If this change is adding new flow

## When you need to add integration test

-   [ ] If components from external libraries are being used to define the flow, e.g. @deriv/components
-   [ ] If it relies on a very specific set of props with no default behavior for the current component.

## Test coverage checklist (for reviewer)

-   [ ] Ensure utility / function has a test case 
-   [ ] Ensure all the tests are passing

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [X] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: packages/components/src/components/money/money.tsx [hunk 1]
```diff
@@ -1,15 +1,30 @@
...
 
-const Money = ({ amount, className, currency = 'USD', has_sign, should_format = true, show_currency = false }) => {
+type TMoneyProps = {
+    amount?: number | string;
+    className?: string;
+    currency?: string;
+    has_sign?: boolean;
+    should_format?: boolean;
+    show_currency?: boolean; // if true, append currency symbol
+};
```
Comment (yauheni-deriv) [inline packages/components/src/components/money/money.tsx]:
```suggestion
type TMoneyProps = {
    amount: number | string;
    className: string;
    currency: string;
    has_sign: boolean;
    should_format: boolean;
    show_currency: boolean; // if true, append currency symbol
};
```

File: packages/components/src/components/money/money.tsx [hunk 2]
```diff
@@ -1,15 +1,30 @@
...
+
+const Money = ({
+    amount = 0,
+    className,
+    currency = 'USD',
+    has_sign,
+    should_format = true,
+    show_currency = false,
+}: TMoneyProps) => {
```
Comment (yauheni-deriv) [inline packages/components/src/components/money/money.tsx]:
```suggestion
}: Partial<TMoneyProps>) => {
```

File: packages/components/src/components/money/money.tsx (hunk 1 above)
Comment (jim-deriv) [reply packages/components/src/components/money/money.tsx:11]: @yauheni-kryzhyk-deriv, many of the `components` that use this `component` don't pass in every prop. I checked `account-has-balance` for example and it doesn't pass in `className`  can you help to confirm this as well?
`payment-agent-withdraw-confirm` would be another

Comment (yauheni-deriv) [reply packages/components/src/components/money/money.tsx]: @jim-deriv I just proposed to to use utility type Partial.
when all props are optional we can use Partial< >
https://www.typescriptlang.org/docs/handbook/utility-types.html

**Review by maryia-deriv — APPROVED** [review]

**Review by maryia-deriv — APPROVED** [review]
👍🏻

**Review by yauheni-deriv — APPROVED** [review]

**Review by niloofar-deriv — APPROVED** [review]
LGTM

**Review by shayan-deriv — APPROVED** [review]

**Pull request merged**